docs/docs/contributing/style_guide.md

```diff
@@ -0,0 +1,106 @@
+## Programming Style Guide
```

Comment:
Moving forward, we should start enforcing stylistic naming conventions as well, and not just formatting. This is an issue across plugins because different plugins use different types of variable names (e.g. camel case, snake case, etc). Especially for member variables because some plugins use an ``m_`` prefix, while others do not, which might make it confusing.

Reply:
I agree. I think as part of this PR I will go through and standardise as much as I can.

Reply:
I have gone through the code base and applied the standards best I could

Reply:
Sorry for the late response. Follow up on this, can we also choose a standard regarding `.h` and `.hpp`? It's fine to leave third-party alone, but for example, we have `plugins/offload_vio/device_tx/video_encoder.h` but `plugins/timewarp_vk/utils/hmd.hpp`. It seems like most files use HPP, so that's what I'm in favor of.

For the future, I would also like to encourage multiple files for our plugin implementations - at the very least, a header file where the plugin class is declared along with member variables and functions. I find it easier to digest what a plugin's doing if I can see all the functions declared in a header file. I think it's also easier to parse member variables from the header file than look for where they are between `public` and `private`, etc.

Reply:
I agree, I will add it to the documentation and make the code changes.

Reply:
changes done